cartographer_ros/cartographer_ros/assets_writer.cc

```diff
@@ -148,25 +151,42 @@ std::unique_ptr<carto::io::PointsBatch> HandleMessage(
   return points_batch;
 }
 
-void RunPipeline(
-    const std::vector<std::unique_ptr<carto::io::PointsProcessor>>& pipeline,
-    const std::vector<std::string>& bag_filenames,
-    const carto::mapping::proto::PoseGraph& pose_graph_proto,
-    const std::string tracking_frame, const std::string& urdf_filename,
-    const bool use_bag_transforms) {
+}  // namespace
+
+AssetsWriter::AssetsWriter(const std::string& pose_graph_filename,
+                           const std::vector<std::string>& bag_filenames,
+                           const std::string& configuration_directory,
+                           const std::string& configuration_basename,
+                           const std::string& urdf_filename,
+                           const std::string& output_file_prefix,
+                           const bool use_bag_transforms)
+    : bag_filenames_(bag_filenames),
+      urdf_filename_(urdf_filename),
```

Comment:
I thought the use-case, that the user wants to use different static transformations for the same setup is quite rare. That's why I added it to the constructor, but I can also move it to the Run() method if you want.

Reply:
My thinking was more that your current code is more complicated than it would need to be. Moving the parameter to `Run()` reduces the amount of state you have to copy and keep, i.e. it is simpler. That it is also more flexible is side effect, but I guess almost all uses we have in mind will create the object and call `Run()` once.